app/models/v2/policy.rb

```diff
@@ -46,5 +48,19 @@ def profile_title
     def ref_id
       attributes['profile__ref_id'] || profile.ref_id
     end
+
+    def update_relationships(new_host_ids, _user)
+      # Remove all hosts assigned to policy then import the new ones
+      # FIXME: scope this after V2::Host is implemented
+      removed = V2::PolicyHost.where(policy: self)&.destroy_all
+      imported = V2::PolicyHost.import_from_policy!(id, new_host_ids)
+
+      # FIXME: this needs to go as each user will have their own counters
+      # rubocop:disable Rails/SkipsModelValidations
+      update_columns(host_count: hosts.count)
+      # rubocop:enable Rails/SkipsModelValidations
```

Comment:
When I wrote "this needs to go" I meant that this has to be not moved to APIv2 😁